src/project_model.rs

```diff
@@ -55,30 +55,32 @@ impl ProjectModel {
                     let v: EncodableResolve = resolve.try_into()?;
                     Some(v.into_resolve(&ws)?)
                 }
-                _ => None
+                _ => None,
             }
         };
-        // then resolve precisely and add overrides
         let mut registry = PackageRegistry::new(ws.config())?;
         let resolve = resolve_with_prev(&mut registry, &ws, prev.as_ref())?;
         let cargo_packages = {
             let ids: Vec<PackageId> = resolve.iter().cloned().collect();
             registry.get(&ids)
         };
-
         let mut pkg_id_to_pkg = HashMap::new();
+        let mut manifest_to_id = HashMap::new();
         let mut packages = Vec::new();
         for (idx, pkg_id) in resolve.iter().enumerate() {
             let pkg = Package(idx);
             pkg_id_to_pkg.insert(pkg_id.clone(), pkg);
             let cargo_pkg = cargo_packages.get(pkg_id)?;
+            let manifest = cargo_pkg.manifest_path().to_owned();
             packages.push(PackageData {
-                manifest: cargo_pkg.manifest_path().to_owned(),
-                lib: cargo_pkg.targets().iter()
+                lib: cargo_pkg
+                    .targets()
+                    .iter()
                     .find(|t| t.is_lib())
-                    .map(|t| t.src_path().to_owned()),
+                    .map(|t| (t.src_path().to_owned(), t.name().replace('-', "_"))),
```

Comment:
Is this mapping performed because we get a Cargo package name and somehow need to map it to a rust crate name, as Racer would expect to? Could you please add a note why the conversion is necessary?

Reply:
> Is this mapping performed because we get a Cargo package name and somehow need to map it to a rust crate name, as Racer would expect to?

Yes.
I'm going to add some comments.